cmd/jujud/agent/model/manifolds.go

```diff
@@ -328,30 +328,30 @@ func IAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 		// it.
 
 		// The undertaker is currently the only ifNotAlive worker.
-		undertakerName: ifNotUpgrading(ifNotAlive(undertaker.Manifold(undertaker.ManifoldConfig{
+		undertakerName: ifNotUpgrading(ifNotAlive(ifCredentialValid(undertaker.Manifold(undertaker.ManifoldConfig{
```

Comment:
3 levels... maybe split over a few lines?

Reply:
As discussed elsewhere, I will be following up this PR with another that will add document workers better. I'll add line breaks there ...